### PR TITLE
LibraryFeature: Rename getChildModel to sidebarModel and make it const

### PR DESCRIPTION
**Pull request by Holzhaus**

### CODE REVIEW

File: src/library/autodj/autodjfeature.h [hunk 1]
```diff
@@ -62,7 +62,7 @@ class AutoDJFeature : public LibraryFeature {
     // The id of the AutoDJ playlist.
     int m_iAutoDJPlaylistId;
     AutoDJProcessor* m_pAutoDJProcessor;
-    TreeItemModel m_childModel;
+    TreeItemModel* m_pSidebarModel;
```
Comment (uklotzde) [inline src/library/autodj/autodjfeature.h:65]: For this purpose we have `parented_ptr`

File: src/library/banshee/bansheefeature.cpp [hunk 2]
```diff
@@ -18,6 +18,7 @@ QString BansheeFeature::m_databaseFile;
 
 BansheeFeature::BansheeFeature(Library* pLibrary, UserSettingsPointer pConfig)
         : BaseExternalLibraryFeature(pLibrary, pConfig),
+          m_pSidebarModel(new TreeItemModel(this)),
```
Comment (uklotzde) [inline src/library/banshee/bansheefeature.cpp:21]: This becomes `make_parented<TreeItemModel>(this)`

File: src/library/autodj/autodjfeature.h (hunk 1 above)
Comment (Holzhaus) [reply src/library/autodj/autodjfeature.h:65]: Doesn't work. Although `QAbstractItemModel` inherits from QObject, I get:

```
/home/jan/Projects/mixxx/src/util/parented_ptr.h: In instantiation of ‘parented_ptr<T>::~parented_ptr() [with T = TreeItemModel]’:
/home/jan/Projects/mixxx/src/library/itunes/itunesfeature.cpp:67:55:   required from here
/home/jan/Projects/mixxx/src/util/parented_ptr.h:31:45: error: no matching function for call to ‘TreeItemModel::parent()’
   31 |         DEBUG_ASSERT(!m_ptr || m_ptr->parent());
      |                                ~~~~~~~~~~~~~^~
```

Comment (uklotzde) [reply src/library/autodj/autodjfeature.h:65]: Caused by conflicting method names in the Qt class hierarchy and a deficiency of `parented_ptr`:

```
DEBUG_ASSERT(!m_ptr || static_cast<const QObject*>(m_ptr)->parent());
```

Please fix it and try again.

**Review by uklotzde — APPROVED** [review]
Thank you! LGTM

**Pull request merged**